ops/charm.py

```diff
@@ -773,7 +773,10 @@ class is mostly for the framework to understand what the charm has defined.
 
     """
 
-    def __init__(self, raw: dict = {}, actions_raw: dict = {}):
+    def __init__(self, raw: dict = None, actions_raw: dict = None):
```

Comment:
Eek! Why did this line of code not jump out to me before?

@jameinel was the intent of the original line to create a global dict that had a life outside of this routine? Or was this just something that got written, and missed comment? 

(I suspect the latter. And since there is only one instance of CharmMeta, I suspect that the change here doesn't matter. But I just wanted to make sure that we're not missing something!)

Reply:
so nothing in here is keeping raw or mutating `raw` or `actions_raw`. So in that sense, it is 'safe' because you don't allow any way to modify it. (eg we don't set self._raw, and we only read from raw,)

But it is an antipattern to use a mutable value as a default and it may be that in the future those constraints won't hold. It certainly wasn't meant to create a mutable default.

Reply:
I touched that line just because I eeked as well a little bit, I'm sure the way the class is used ATM is 'safe', but I've seen such horrible bugs arise out of mutable kwarg defaults that I thought -- let's be safe and get rid of it while we're at it